bebot/src/bebot.cpp

```diff
@@ -0,0 +1,47 @@
+#include <bebot/bebot.hpp>
+#include <Eigen/Dense>
+#include <iostream>
+
+using namespace bebot;
+
+BeBOT::BeBOT(const Eigen::Ref<Eigen::MatrixXd>& cpts) {
+    set_cpts(cpts);
+    set_t0(0.0);
+    set_tf(1.0);
+}
+
+BeBOT::BeBOT(const Eigen::Ref<Eigen::MatrixXd>& cpts, const double tf) {
+    set_cpts(cpts);
+    set_t0(0.0);
+    set_tf(tf);
+}
+
+BeBOT::BeBOT(const Eigen::Ref<Eigen::MatrixXd>& cpts, const double t0, const double tf) {
+    set_cpts(cpts);
+    set_t0(t0);
+    set_tf(tf);
+}
+
+void BeBOT::set_cpts(const Eigen::Ref<Eigen::MatrixXd>& cpts) {
```

Comment:
More descriptive naming is good. set_control_points isn't that many more characters. In general, we try to avoid writing C++ like it was C, and the horrible idmtcNmng_cnvns that exist in C shouldnt happen in C++